dvc/repo/tree.py

```diff
@@ -111,6 +146,15 @@ def walk(self, top, topdown=True):
         for out in outs:
             trie[out.path_info.parts] = out
 
+            if out.is_dir_checksum and (self.fetch or self.stream):
+                # will pull dir cache if needed
```

Comment:
So it doesn't pull the whole dir (i mean with the files contents), yet, right? That will be introduced later? Just making sure I follow :slightly_smiling_face:

Reply:
Yeah, that will come w/the next PR (for saving a tree to cache). For now I just wanted to get `walk()` working properly in this PR.